### ISSUE TITLE
[Bug]: "soft keyboard only if no hardware keyboard" doesn't always work

### ISSUE THREAD

**Issue by amaramth** [termux/termux-app#3450]
### Problem description

Even with "soft keyboard only if no hardware keyboard" enabled, sometimes the soft keyboard shows up anyway. Sometimes it's just the auto-complete area of the soft keyboard, other times it's the entire thing.

### Steps to reproduce the behavior.

Not known.

### What is the expected behavior?

"soft keyboard only if no hardware keyboard" removes the soft keyboard if it does show up.

### System information

* Termux application version: 0.118.0
* Android OS version: 8.0.0
* Device model: F5321

**Comment by amaramth** [comment]
yes, the soft keyboard goes esc / - home ij end etc. are you able to disable it when you want to, and it keeps popping back up, or are you unable to disable it?

**Comment by Rendam180** [comment]
> ### Problem description
> 
> Even with "soft keyboard only if no hardware keyboard" enabled, sometimes the soft keyboard shows up anyway. Sometimes it's just the auto-complete area of the soft keyboard, other times it's the entire thing.
> 
> I filed this bug because keeping the soft keyboard closed prevents some other termux bugs that have more impact.
> 
> ### Steps to reproduce the behavior.
> 
> Not known.
> 
> ### What is the expected behavior?
> 
> "soft keyboard only if no hardware keyboard" removes the soft keyboard if it does show up.
> 
> ### System information
> 
> * Termux application version: 0.118.0
> * Android OS version: 8.0.0
> * Device model: F5321

Hello 
https://github.com/user-attachments/assets/216ca2b4- [ ] - - [ ] [ ] - [ ] -e468-46b9-97eb-e38880ec29b7

**Comment by amaramth** [comment]
oy i confused myself. this bug is about the android soft keyboard which is not in your screenshot. that's the termux soft keyboard, defined in `~/.termux/termux.properties`. the default key for toggling it is `[volume up button] [q]` pressed at the same time.

**Comment by ssmus** [comment]
If this bug is about the Android keyboard, I just found what I needed to do on my device to solve the issue. Hope it helps.

It's an Android setting, not a Termux setting.

Settings > System > Keyboard > On-screen keyboard > Gboard > Physical keyboard > Toolbar > Show on-screen keyboard ("Show on-screen keyboard while using physical keyboard")

Note that you have to go into the Physical Keyboard settings within the Gboard (on-screen keyboard) settings. Not the Physical Keyboard settings in the System/Keyboard settings.
